src/signals/settings/roles/containers/RoleFormContainer/components/RoleForm/index.js

```diff
@@ -27,6 +27,17 @@ const StyledInput = styled(Input)`
 export const RoleForm = ({ role, permissions, onPatchRole, onSaveRole, readOnly }) => {
   const formRef = useRef(null);
   const { isValid, validate, errors, event } = useFormValidation(formRef);
+  const [rolePermissions, setRolePermissions] = useState(role.permissions);
```

Comment:
Since the state value `rolePermissions` contains an array of objects, wouldn't it be better to use `useReducer` for this instead of `useState`?

Reply:
While the array is used as a whole and there is no different way of modifying the data but replacing (no different event-state relation), I don't see the point.